### PR TITLE
[ELY-2084] Split Elytron SSL into its own subsystem and layer

### PR DESCRIPTION
**Pull request by jessicarod7**
Isolates TLS code from Elytron subsystem. PR is not ready to be merged yet. Continuing from #1678.

Related issues:
- https://issues.redhat.com/browse/ELY-2084
- https://issues.redhat.com/browse/WFCORE-5279
- https://issues.redhat.com/browse/EAP7-1564

### CODE REVIEW

File: .vscode/configurationCache.log [hunk 1]
```diff
@@ -0,0 +1 @@
+{"buildTargets":[],"launchTargets":[],"customConfigurationProvider":{"workspaceBrowse":{"browsePath":[],"compilerArgs":[]},"fileIndex":[]}}
```
Comment (fjuma) [inline .vscode/configurationCache.log]: The .vscode files should be removed.

File: ssl/base/src/main/java/org/wildfly/security/ssl/DomainlessSSLContextBuilder.java [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+ * <ul>
+ *     <li>The cipher suite selector defaults to {@link CipherSuiteSelector#openSslDefault()}</li>
+ *     <li>The protocol suite selector defaults to {@link ProtocolSelector#DEFAULT_SELECTOR}</li>
+ *     <li>The "require client authentication" flag defaults to {@code false}</li>
+ *     <li>The provider supplier defaults to {@link Security#getProviders() Security::getProviders}</li>
+ * </ul>
+ *
+ * @author <a href="mailto:david.lloyd@redhat.com">David M. Lloyd</a>
+ */
+public final class DomainlessSSLContextBuilder {
```
Comment (fjuma) [inline ssl/base/src/main/java/org/wildfly/security/ssl/DomainlessSSLContextBuilder.java]: Just trying to think of some possible names here to make it more clear what this class is about. Perhaps `BaseSSLContextBuilder`?

Comment (fjuma) [reply ssl/base/src/main/java/org/wildfly/security/ssl/DomainlessSSLContextBuilder.java]: Just a general note, for all the files in this PR that have been renamed / copied, we'll need to check if we've pulled in the latest updates to these files (just in case anything has changed since the PR was originally created).

File: ssl/base/src/main/java/org/wildfly/security/ssl/ElytronMessages.java [hunk 3]
```diff
@@ -111,7 +110,7 @@ interface ElytronMessages extends BasicLogger {
     CertificateException emptyChainNotTrusted();
 
     @Message(id = 4003, value = "Certificate not trusted due to realm failure for principal [%s]")
-    CertificateException notTrustedRealmProblem(@Cause RealmUnavailableException e, Principal principal);
+    CertificateException notTrustedRealmProblem(@Cause Throwable e, Principal principal);
```
Comment (skyllarr) [inline ssl/base/src/main/java/org/wildfly/security/ssl/ElytronMessages.java]: @cam-rod why was this changed?

File: .vscode/configurationCache.log (hunk 1 above)
Comment (jessicarod7) [reply .vscode/configurationCache.log]: oh yeah, removed

File: ssl/base/src/main/java/org/wildfly/security/ssl/ElytronMessages.java (hunk 3 above)
Comment (jessicarod7) [reply ssl/base/src/main/java/org/wildfly/security/ssl/ElytronMessages.java]: That wasn't my change, it was from before. My guess is it was to remove the dependency on `org.wildfly.security:wildfly-elytron-auth-server` -  that module can actually be deleted from the `ssl-base` POM because it has no other uses.

Comment (jessicarod7) [reply ssl/base/src/main/java/org/wildfly/security/ssl/ElytronMessages.java]: Took another look and it looks like a number of error messages are duplicated between `ssl-base` and `ssl-domain`. This message isn't used in `ssl-base`, so I'll just remove it and the other duplicates.

**Comment by jessicarod7** [comment]
Removed unused dependencies from `ssl-base` (include `wildfly-elytron-auth` and `wildfly-elytron-auth-server`), along with ElytronMessages duplicated between the SSL modules.

File: ssl/base/src/main/java/org/wildfly/security/ssl/DomainlessSSLContextBuilder.java (hunk 2 above)
Comment (jessicarod7) [reply ssl/base/src/main/java/org/wildfly/security/ssl/DomainlessSSLContextBuilder.java]: yeah, I like that name better. I'll also update both Javadocs to explain the difference.

Comment (jessicarod7) [reply ssl/base/src/main/java/org/wildfly/security/ssl/DomainlessSSLContextBuilder.java]: Renamed to `BaseSSLContextBuilder`, and also reviewed the files for new changes. There were a couple in the test cases, but since they are identical between the modules, I just created a test-jar for `wildfly-elytron-ssl-base` and called the code directly (similar to what I did in wildfly/wildfly-core#5561).

**Comment by fjuma** [comment]
Going to close this one for now until we are ready to pick up working on it again.
